### PR TITLE
Add docs search

### LINKED ISSUE #89: Add searchbox to generated documentation
We should be able to generate documentation for all the libraries in the package set (and/or just the libraries used in the project), in a similar way to what `stack haddock` does for Haskell projects.

The use-cases for this are multiple and useful:
- get specialized documentation for your project without having to rely on Pursuit (because you might have dependencies not from the package set and/or on Pursuit)
- more generally, allow an easy way to have an offline Pursuit
- generate a "static Pursuit" for every package set in Spacchetti and host that alongside the package set, which is effectively the [Stackage](https://www.stackage.org/) approach.  
  Note that this would obsolete #27, as it would entirely remove the "where do I upload my library" problem → you would just add it to a Spacchetti package-set, and have it "published" in the next Spacchetti release

How to achieve this:
- it is possible to generate html docs for the whole package-set by running a `purs docs` command.  
  Justin had a good start with [acme-spago](https://github.com/justinwoo/acme-spago). He included a spago project in there, but if this is integrated in spago itself, we wouldn't need it (as we would just generate that info on the fly)
- then we'd need a way to search in the existing docs. Since the whole thing needs to be static files (i.e. no backend), we should have a frontend-side search). A simple way to do this would be to generate markdown docs and then index them with something like [Lunr](https://lunrjs.com/) (@justinwoo suggested scraping the externs JSON files, but I'm not sure what that would entail, so probably worth detailing more)

### PR DESCRIPTION
**Pull request by klntsky**
The binary size is now 48MB.

Some more features to consider:

- Add a CLI flag that disables docs patching when running `spago docs`.
- Add ability to skip compilation when running `spago search`.
- Pin JS files by hash

Fix #89

### CODE REVIEW

File: app/Spago.hs [hunk 1]
```diff
@@ -229,6 +231,11 @@ parser = do
       , Docs <$> docsFormat <*> sourcePaths <*> depsOnly
       )
 
+    search =
+      ( "search"
+      , "Run the search engine."
```
Comment (f-f) [inline app/Spago.hs]:
```suggestion
      , "Start a search REPL to find definitions matching names and types"
```

File: src/Spago/Templates.hs [hunk 2]
```diff
@@ -25,3 +25,15 @@ gitignore = $(embedFileUtf8 "templates/gitignore")
 
 bowerJson :: B.ByteString
 bowerJson = $(embedFile "templates/bower.json")
+
+docsSearchApp :: T.Text
+docsSearchApp =
+  $(embedURLWithFallback
+    "https://github.com/spacchetti/purescript-docs-search/releases/download/v0.0.4/docs-search-app.js"
```
Comment (f-f) [inline src/Spago/Templates.hs]: Let's factor out the tag in a top level definition, as:
- we might want to use it (e.g. in `spago version`)
- it's easier to upgrade the templates

**Comment by f-f** [comment]
@klntsky I'll merge now so we can include this in today's release. Thank you for doing this! 🙂

File: src/Spago/Templates.hs (hunk 2 above)
Comment (klntsky) [reply src/Spago/Templates.hs]: > ‘docsSearchVersion’ is used in a top-level splice, quasi-quote, or annotation,
>        and must be imported, not defined locally

To which module should I put it?

Comment (f-f) [reply src/Spago/Templates.hs]: @klntsky how about `Prelude`?

**Comment by googleson78** [comment]
This is awesome! Thanks!

**Pull request merged**